HeartbeatLogging/Sources/HeartbeatsPayload.swift

```diff
@@ -84,21 +84,20 @@ extension HeartbeatsPayload: HTTPHeaderRepresentable {
   /// Returns a processed payload string intended for use in a HTTP header.
   /// - Returns: A string value from the heartbeats payload.
   public func headerValue() -> String {
-    // TODO(ncooke3): Return a empty payload instead of empty string.
-    if userAgentPayloads.isEmpty {
-      return ""
-    }
-
```

Comment:
1. Previously, when the `headerValue` is requested from an empty payload, it would return an empty string. This empty string was not process correctly by backend. So instead, we treat an empty payload like any other payload and compress and zip it.

Reply:
When is there an empty payload? And why does it need to be sent?

Reply:
> When is there an empty payload?

Let's say the app is running and heartbeats are flushed to be added to a product request. Then, heartbeats are again flushed for another product request. This second flushing will result in an *empty payload*. Heartbeats are removed upon retrieval to avoid being retrieved (and potentially sent!) twice.

> And why does it need to be sent?

I guess technically it doesn't.

My rationale for this is that it simplifies things for the client. I believe this is what other platforms did as well.

Current usage by the client would be like:
```swift
let heartbeatsPayload = heartbeatController.flush()
let heartbeatsPayloadHeaderString = heartbeatsPayload.headerValue()
// add `heartbeatsPayloadHeaderString` to request
```

If flushing the heartbeat controller returned `nil` when there were no heartbeats to send, things would instead look like:
```swift
let heartbeatsPayload = heartbeatController.flush()
if let heartbeatsPayload = heartbeatsPayload {
  let heartbeatsPayloadHeaderString = heartbeatsPayload.headerValue()
  // add `heartbeatsPayloadHeaderString` to request
}
```
Keeping in mind the above code would be in objc for our product SDKs.

Reply:
Oh, I think the point I was missing is that the client is going to send something to the backend either way. And we want at least a placeholder for the heartbeat data.